stackify/transport/agent/agent_socket.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import os
+import retrying
+import requests_unixsocket
+
+
+internal_logger = logging.getLogger(__name__)
+
+
+class AgentSocket(object):
+    SOCKET_LOG_FILE = 'http+unix://%2Fusr%2Flocal%2Fstackify%2Fstackify.sock'
+    SOCKET_SCHEME = 'http+unix://'
+
+    def __init__(self):
+        self._session = requests_unixsocket.Session()
+
+    def _post(self, url, payload):
+        if not url.startswith(self.SOCKET_SCHEME):
+            url = os.path.join(self.SOCKET_LOG_FILE, url.lstrip('/'))
+
+        internal_logger.debug('Request URL: {}'.format(url))
+        internal_logger.debug('POST data: {}'.format(payload))
+
+        headers = {
+            'Content-Type': 'application/x-protobuf',
+        }
+
+        try:
+            response = self._session.post(url, payload, headers=headers)
+            internal_logger.debug('Response status: {}'.format(response.status_code))
+            return response
+        except Exception as e:
+            internal_logger.debug('HTTP UNIX Socket domain exception: {}.'.format(e))
+            raise
```

Comment:
does the raising of this exception cause any problems w/ the application being logged ?

Reply:
doesn't cause any problem I guess since we are catching the exception and does internal loggings in the transport part.